src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -134,12 +139,16 @@ public void setPeriod(String period) {
     public void visitJavadocToken(DetailNode ast) {
         String firstSentence = getFirstSentence(ast);
         final int endOfSentence = firstSentence.lastIndexOf(period);
-        if (endOfSentence == -1) {
-            if (!isOnlyInheritDoc(ast)) {
-                log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
-            }
+        final String summaryDoc = getSummarySentence(ast);
```

Comment:
I will not be able to merge both methods into one because in getSummarySentence i am trying to fetch details upto first period, and in getFirstSentence we will be getting whole first sentence.I have been trying to merge these 2 methods but is not working because purpose of both methods are totally different that is what the reason why i had to introduce my own method.